### ISSUE TITLE
IndexOutOfRangeException (incorrect Head) after using Get(count) to empty Full buffer

### ISSUE THREAD

**Issue by dennisverheijen** [cyotek/Cyotek.Collections.Generic.CircularBuffer#17]
I encountered this problem in v1.1.0 after using the `Get(int count)` overload to read a buffer (that has AllowOverwrite=false) 1 byte at a time when the buffer is completely full. When using the `Get<T>()` instead, it works fine. The problem seems to be an incorrect Head value after the Get() loop, see comment inside the code.

See this code that shows the problem:
```c#
using Cyotek.Collections.Generic;

namespace ConsoleApp12
{
    class Program
    {
        static void Main(string[] args)
        {
            Test(true);
            Test(false);
        }

        const int Size = 10;
        private static void Test(bool mode)
        {
            var buffer = new CircularBuffer<byte>(Size, false);

            var fullSizeInput = new byte[Size];
            buffer.Put(fullSizeInput);

            while (!buffer.IsEmpty)
            {
                if (mode)
                {
                    var val = buffer.Get();
                }
                else
                {
                    var array = buffer.Get(1);
                }
            }

            // At this point, for mode = true, Head=0, Tail=0, Size=0
            // At this point, for mode = false, Head=10 (!), Tail=0, Size=0

            var smallInput = new byte[1];
            buffer.Put(smallInput);

            // This throws and Index out of range exception for mode=false
            var fail = buffer.Peek();
        }

    }
}
```

**Comment by cyotek** [comment]
Hello,

Thank you for the bug report and reproduction! I've published a new 1.2.0-alpha.1 package with a proposed fix. If you're able to give this a whirl, feedback would be welcome, otherwise I'll do some more poking and try and get a final release build in the next few days.

Regards;
Richard Moss

**Comment by dennisverheijen** [comment]
Many thanks for your fast response. This fixes my specific issue! My other unit tests are also still green.
